app/client/src/store/mutations.ts

```diff
@@ -93,5 +102,15 @@ export default {
         // indication of submitted status in the response for all projects including those which had never been
         // submitted and will need to update this line accordingly.
         state.submitted = true;
+    },
+    projectRenamed(state: RootState, payload: RenameProjectPayload) {
+        const { projectId, name } = payload;
+        const savedProject = state.savedProjects.find((p) => p.id === projectId);
+        if (savedProject) {
+            savedProject.name = name;
+        }
+        if (state.projectId === projectId) {
```

Comment:
Yeah, because you might be renaming a non-current project in the project list. 

The previous check *should* always be true, but Typescript considers that the result of the find could be null so we have to do something here. Could just assert non-null instead.